### ISSUE TITLE
App being rejected due to the the use of the 'hide' method.

### ISSUE THREAD

**Issue by BlaineOmega** [jdg/MBProgressHUD#373]
I'm submitting an app to Apple and they are rejecting the app due to the use of the method name 'hide'. 

Here is the letter they sent me:

> Dear developer,
> 
> We have discovered one or more issues with your recent delivery for "yourAppName". To process your delivery, the following issues must be corrected:
> 
> Non-public API usage:
> 
> The app references non-public selectors in youApp: hide:, imageWithName:, setLabelText:
> If method names in your source code match the private Apple APIs listed above, altering your method names will help prevent this app from being flagged in future submissions. In addition, note that one or more of the above APIs may be located in a static library that was included with your app. If so, they must be removed.
> 
> If you think this message was sent in error and that you have only used Apple-published APIs in accordance with the guidelines, send the app's nine-digit Apple ID, along with detailed information about why you believe the above APIs were incorrectly flagged, to appreview@apple.com. For further information, visit the Technical Support Information page.
> 
> Once these issues have been corrected, you can then redeliver the corrected binary.
> 
> Regards,
> 
> The App Store team

It should be noted that we aren't using the methods imageWithName:, setLabelText:.

**Comment by yas375** [comment]
I've just looked through the source of the library and I can't find the usages of methods `hide:` or `imageWithName:` at all. But there is a deprecated `setLabelText:`. Looks like same to #317.

It's definitely not a private api use. Maybe try to contact Apple and point them to the source of this library?

**Comment by matej** [comment]
MBProgressHUD doesn't use any private API. You'll have to check your code and other libraries.

**Comment by BlaineOmega** [comment]
I know you don't use any private API's. However, you use method names, like hide, that are private API method names. This makes the build get flagged by Apple. You should know that when I removed your product, and used another one, I did not have my builds rejected.

**Comment by matej** [comment]
Interesting. I have a couple of apps using it and never got a private API rejection.

**Comment by yas375** [comment]
> If you think this message was sent in error and that you have only used Apple-published APIs in accordance with the guidelines, send the app's nine-digit Apple ID, along with detailed information about why you believe the above APIs were incorrectly flagged, to appreview@apple.com.

@BlaineOmega have you tried this? This is clearly the thing I would do first...

**Comment by BlaineOmega** [comment]
@yas375 Of course. It was the first thing that I did. However, they have not got back to me. To be honest, I'm not interested in waiting on Apple to get back to me. @matej I didn't have an issue to begin with. It only recently started happening. Anyways, what appears to be happening is that they were flagging my app based on the 'hide' method name being there. I'm sure that once they look at things, they will approve, I just don't have time. I figured I would mention all this in case it becomes a bigger issue in the future.

**Comment by azav** [comment]
I just checked our shipping app that is in the App Store and it sure has code in it that uses MBProgressHud's hide method. In fact I made a wrapper and in my wrapper class, I have my own show and hide methods.

I'm not sure that Apple's complaint is correct.  Now, I can't see your code so I can't tell if there is some funky usage on your part either, but this sure seems strange.

In any case, this is a simple case of, "just rename the method and resubmit", if Apple wants to be this way, right?
